101-machine-learning-create/azuredeploy.json

```diff
@@ -2,257 +2,641 @@
-  },
-  "resources": [
-      {
-          "condition": "[equals(parameters('storageAccountNewOrExisting'), 'new')]",
-          "type": "Microsoft.Storage/storageAccounts",
-          "apiVersion": "2019-04-01",
-          "name": "[parameters('storageAccountName')]",
-          "location": "[parameters('location')]",
-          "sku": {
-              "name": "[parameters('storageAccountType')]"
-          },
-          "kind": "StorageV2",
-          "properties": {
-              "encryption": {
-                  "services": {
-                      "blob": {
-                          "enabled": true
-                      },
-                      "file": {
-                          "enabled": true
-                      }
+    },
+    {
+      "condition": "[equals(parameters('privateEndpointType'), 'AutoApproval')]",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2017-05-10",
+      "name": "[concat('PrivateDns-', variables('privateDnsGuid'))]",
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/privateEndpoints', concat(parameters('workspaceName'), '-PrivateEndpoint'))]"
+      ],
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+          "contentVersion": "1.0.0.0",
+          "resources": [
+            {
+              "type": "Microsoft.Network/privateDnsZones",
+              "apiVersion": "2018-09-01",
+              "name": "privatelink.api.azureml.ms",
+              "location": "global"
+            },
+            {
+              "type": "Microsoft.Network/privateDnsZones/virtualNetworkLinks",
+              "apiVersion": "2018-09-01",
+              "name": "[concat('privatelink.api.azureml.ms', '/', uniqueString(resourceId('Microsoft.Network/virtualNetworks', variables('vnetName'))))]",
+              "location": "global",
+              "dependsOn": [
+                "privatelink.api.azureml.ms"
+              ],
+              "properties": {
+                "virtualNetwork": {
+                  "id": "[resourceId('Microsoft.Network/virtualNetworks', variables('vnetName'))]"
+                },
+                "registrationEnabled": false
+              }
+            },
+            {
+              "apiVersion": "2017-05-10",
+              "name": "[concat('EndpointDnsRecords-', variables('privateDnsGuid'))]",
+              "type": "Microsoft.Resources/deployments",
+              "dependsOn": [
+                "privatelink.api.azureml.ms"
+              ],
+              "properties": {
+                "mode": "Incremental",
+                "templatelink": {
+                  "contentVersion": "1.0.0.0",
+                  "uri": "https://network.hosting.portal.azure.net/network/Content/4.13.392.925/DeploymentTemplates/PrivateDnsForPrivateEndpoint.json"
```

Comment:
all files needed for deployment must be included as part of the sample/repo

Reply:
Refactored this section. Removed the files reference